### ISSUE TITLE
Generated adapter doesn't work with number nullable types as generics

### ISSUE THREAD

**Issue by entibaldi** [square/moshi#671]
Adapter generation for classes like the following doesn't work.
```
@JsonClass(generateAdapter = true)
data class Foo(
    @Json(name = "map") val map: Map<String, Float?>
)
```
The reason is that generated adapter uses the following type
```
Types.newParameterizedType(Map::class.java, String::class.java, Float::class.java)
```
instead of 
```
Types.newParameterizedType(Map::class.java, String::class.java, java.lang.Float::class.java)
```
resulting in runtime parsing exception `java.lang.IllegalArgumentException: Unexpected primitive float. Use the boxed type.`

**Comment by JakeWharton** [comment]
Fixed by #627?

**Comment by entibaldi** [comment]
Looks like it will be fixed in 1.7
Closing.
